packages/flutter/lib/src/material/floating_action_button_location.dart

```diff
@@ -49,27 +73,229 @@ abstract class FloatingActionButtonLocation {
   /// const constructors so that they can be used in const expressions.
   const FloatingActionButtonLocation();
 
-  /// End-aligned [FloatingActionButton], floating at the bottom of the screen.
+  /// Start-aligned [FloatingActionButton], floating over the transition between
+  /// the [Scaffold.appBar] and the [Scaffold.body].
   ///
-  /// This is the default alignment of [FloatingActionButton]s in Material applications.
-  static const FloatingActionButtonLocation endFloat = _EndFloatFloatingActionButtonLocation();
+  /// To align a floating action button with [CircleAvatar]s in the
+  /// [ListTile.leading] slots of [ListTile]s in a [ListView] in the [Scaffold.body],
+  /// use [miniStartTop] and set [FloatingActionButton.mini] to true.
+  ///
+  /// This is unlikely to be a useful location for apps that lack a top [AppBar]
+  /// or that use a [SliverAppBar] in the scaffold body itself.
+  static const FloatingActionButtonLocation startTop =
+      _StandardFloatingActionButtonLocation(
+        _startOffsetX,
+        _topOffsetY,
+        'startTop',
+      );
+
+  /// Start-aligned [FloatingActionButton], floating over the transition between
+  /// the [Scaffold.appBar] and the [Scaffold.body], optimized for mini floating
+  /// action buttons.
+  ///
+  /// This is intended to be used with [FloatingActionButton.mini] set to true,
+  /// so that the floating action button appears to align with [CircleAvatar]s
+  /// in the [ListTile.leading] slot of a [ListTile] in a [ListView] in the
+  /// [Scaffold.body].
+  ///
+  /// This is unlikely to be a useful location for apps that lack a top [AppBar]
+  /// or that use a [SliverAppBar] in the scaffold body itself.
+  static const FloatingActionButtonLocation miniStartTop =
```

Comment:
as a user I would not expect to have to use two different locations for `startTop` for mini fab and regular fab.

But I know that this pattern already existed, so I'm sure there's a requirement that makes it so that miniFab have to have their own `FloatingActionButtonLocation`s?

Reply:
Material recommends aligning mini FABs with leading circle avatars (see [Material.io ¶ FAB § Anatomy](https://material.io/components/buttons-floating-action-button/#anatomy)). Also see the 2nd image in this PR for an example.

`startTop` is optimized for regular FABs, so if we use it on mini FABs, they will not align with leading/trailing circle avatars. From what I understand, if there are no circle avatars on the page, it should be fine for us to use `startTop` instead of `miniStartTop`.

So **I guess** using `miniStartTop` with mini FABs is recommended, but optional.